api/tests/unit/features/multivariate/test_unit_multivariate_models.py

```diff
@@ -38,7 +39,7 @@ def test_multivariate_feature_option_get_delete_log_message_for_deleted_feature(
 ):
     # Given
     mvfo = MultivariateFeatureOption.objects.create(feature=feature, string_value="foo")
-    feature.delete()
+    mvfo.feature.delete()
```

Comment:
It's because this [hook](https://github.com/Flagsmith/flagsmith/pull/2358/files#diff-e48d5e2e19f8c9f0d2d868206fe36a2780573ee3e6828bad61cf0ee4c304125fR64) populates the `self.feature` which happens before the feature is deleted

Reply:
(sorry, I think I was a bit too quick with my comment 😄 )

Reply:
> I think we've done a bit of both here, right? We don't actually need the `mvfo.refresh_from_db()` if we're deleting the feature on the `mvfo` object directly.
> 
> The code should either be:
> 
> ```python
> # <comment explaining why we refresh from db>
> mvfo.refresh_from_db()
> feature.delete()
> ```
> 
> or
> 
> ```python
> # <comment explaining why we delete the feature from the mvfo object directly>
> mvfo.feature.delete()
> ```
> 
> Seems like you've done a little bit of both here which adds a bit of extra confusion?

:facepalm: